kml/kml_tests/gpx_tests.cpp

```diff
@@ -0,0 +1,158 @@
+#include "testing/testing.hpp"
+#include "map/bookmark_helpers.hpp"
+#include "kml/serdes_gpx.hpp"
+#include "coding/string_utf8_multilang.hpp"
+#include "geometry/mercator.hpp"
+#include "platform/platform.hpp"
+
+auto const kDefaultCode = StringUtf8Multilang::kDefaultCode;
+
+kml::FileData loadGpxFromString(const std::string& content) {
+  kml::FileData dataFromText;
+  try
+  {
+    const char * input = content.c_str();
+    kml::DeserializerGpx des(dataFromText);
+    MemReader reader(input, strlen(input));
+    des.Deserialize(reader);
+    return dataFromText;
+  }
+  catch (kml::DeserializerGpx::DeserializeException const & exc)
+  {
+    TEST(false, ("Exception raised", exc.what()));
+  }
+}
+
+kml::FileData loadGpxFromFile(std::string file) {
+  auto fileName = GetPlatform().TestsDataPathForFile(file);
+  std::ifstream t(fileName);
+  std::stringstream buffer;
+  buffer << t.rdbuf();
```

Comment:
Can FileReader::ReadAsString() be used here?

Reply:
Fixed